### PR TITLE
Parser Cache Fix

### PR DESCRIPTION
**Pull request by ArquintL**
Currently, the parser cache caches parse ASTs. However, it does not take the `specOnly` flag into account, meaning that there could be the situation of parsing a package first with `specOnly` set to true (because it's imported in a different package) and later parsing the same package with `specOnly` turned off. In this case, the cache would incorrectly return an AST with abstract functions & methods.
This PR fixes the cache to keep the ASTs for a given file and different  `specOnly` flags separate by including the `specOnly` flag in the key used for cache lookups.

### CODE REVIEW

File: src/main/scala/viper/gobra/frontend/Parser.scala [hunk 1]
```diff
@@ -53,9 +53,9 @@ object Parser {
   // cache maps a key (obtained by hasing file path and file content) to the parse result
   private var sourceCache: Map[SourceCacheKey, (Either[Vector[ParserError], PProgram], Positions)] = Map.empty
 
-  /** computes the key for caching a particular source. This takes the name as well as content into account */
-  private def getCacheKey(source: Source): SourceCacheKey = {
-    val key = source.name ++ source.content
+  /** computes the key for caching a particular source. This takes the name, the specOnly flag, and the file's content into account */
```
Comment (jcp19) [inline src/main/scala/viper/gobra/frontend/Parser.scala:56]:
```suggestion
  /** computes the key for caching a particular source. This takes the name, the specOnly flag, and the file's contents into account */
```

Comment (ArquintL) [reply src/main/scala/viper/gobra/frontend/Parser.scala:56]: I would keep it singular as a file only has one content

Comment (jcp19) [reply src/main/scala/viper/gobra/frontend/Parser.scala:56]: Content and contents have slightly different meanings. https://getproofed.com/writing-tips/word-choice-content-vs-contents/

Comment (ArquintL) [reply src/main/scala/viper/gobra/frontend/Parser.scala:56]: Thanks, I was not aware of this. Sorry

**Pull request merged**